lib/split/experiment_catalog.rb

```diff
@@ -13,10 +13,7 @@ def self.all_active_first
     end
 
     def self.find(name)
-      Split.cache(:experiment_catalog, name) do
```

Comment:
I kinda like keeping caching at the Catalog level.  Like the low-level experiment always loads, so you have a path to get the latest, if you really want it.  Like dashboard could use the low-level always load, but there's a higher level that caches for frequent operations.  No idea if that fits the overall arch, but is how I'm thinking about it.

Reply:
The functionality is not affected by this change....the change still means `ExperimentCatalog#find` call will be cached. The dashboard's low level cache, if it's bypassing `ExperimentCatalog` will still continue to load the uncached version.

There are two ways we can think about it:
- `Experiment` class accesses the directory of experiments (`ExperimentCatalog`) to know what the current experiment config is. I personally feel this is a reasonable relationsip, but you mentioned `Experiment` class using `ExperimentCatalog` feels like child class is calling its parent.
- Add a factory method to Experiment, which is cached, and have both `ExperimentCatalog` and `Experiment` class  use this. 

I don't know if there is a third solution....but as I mentioned before, we have to use the same cache namespace to return the experiment config for both calling locations.

Reply:
I think the refactor you did of the `find` method down to the `Experiment` model is good, and resolves my concern 👍